pkg/common/purl/purl.go

```diff
@@ -31,17 +32,26 @@ func ParsePackageUrl(purl string) (*purlResponseWrapper, error) {
 	}
 
 	if instance.Version == "" || instance.Version == "latest" {
+		ui.PrintMsg("Resolving latest version")
```

Comment:
Don't use `ui.PrintMsg` in library code. There is a specific reason why UI is in `internal` package. We want it to be internal to `vet` and specific to `vet`. Lib code is reusable, in `vet` and outside `vet`. Use `logger.Debugf/Infof` if required. `UI` functions should be used only in `cmd` files to ensure every part of the system can be re-used outside `vet` as well.

Imagine we build a `vet-server` that uses `vet` as a Go module.

Reply:
Also it seems to me we should not resolve version here. PURL parser is only for parsing. Such low level library should not automatically handle anything. Its more of "application logic". This is why we handle the case of GitHub Action commit sha resolution at `cmd` level because thats app logic.

Can we avoid doing version resolution or any "automatic" stuff in `pkg/common`. To support `vet scan --purl`, look at PURL reader in `pkg/readers`